gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -327,13 +326,20 @@ void OGRFlatGeobufLayer::Create() {
         return;
     }
 
-    Rect extent = calcExtent(m_featureItems);
+    NodeItem extent = calcExtent(m_featureItems);
     auto extentVector = extent.toVector();
 
     writeHeader(m_poFp, m_featuresCount, &extentVector);
 
     CPLDebug("FlatGeobuf", "Sorting items for Packed R-tree");
     hilbertSort(m_featureItems);
+    CPLDebug("FlatGeobuf", "Calc new feature offsets");
+    auto featureOffset = 0;
```

Comment:
a bit dangerous here to use auto for variable declaration. I guess this resolves to int, and not as uint64 as likely expected ? Did you test with > 4 GB files ?

Reply:
Changed. I haven't tested with > 4 GB files but would like too. It's mostly due to it being time consuming to find and convert a suitable source.

Reply:
maybe from there: https://download.osmdata.xyz/

Reply:
Made some tests using the waterways dataset. Got a 26 GB big file and it works well. Though it did seem a bit large.. which looks like it's explained by the packages from osmdata.xyz containing empty strings instead of null for unset attributes. Makes me think it would be nice with a generic option to convert empty strings to null for ogr2ogr?

Reply:
>  a generic option to convert empty strings to null for ogr2ogr?

something that is regularly asked for indeed. Want to tackle that ? -emptyStrAsNull ? We have a EMPTY_STRING_AS_NULL open option for the CSV driver